File: 2024/10 - October/12. Two Smallests in Every Subarray/Solution.java
class Solution 
{
    public int pairWithMaxSum(int[] arr) 
    {
        int n = arr.length;
        if(n < 2) return -1;
        
        int maxsum = -1;
        
        for(int i=0; i<n-1; ++i)
        {
            int curr = arr[i] + arr[i+1];
            maxsum = Math.max(maxsum, curr);
        }
        
        return maxsum;
    }
}

File: 2024/10 - October/12. Two Smallests in Every Subarray/Solution.py
class Solution:
    def pairWithMaxSum(self, arr):
        if len(arr) < 2:
            return -1
        
        maxsum = -1
        for i in range(len(arr)-1):
            curr = arr[i] + arr[i+1]
            maxsum = max(maxsum, curr)
        
        return maxsum
